### PR TITLE
Added: Linux ioctls

### PR DESCRIPTION
**Pull request by cecton**

### CODE REVIEW

**Comment by cecton** [comment]
@mmstick I have added the ioctls like this. Would you mind giving a review? Let met know what you think :)

**Comment by mmstick** [comment]
It may be useful to expose the functionality through a convenient safe function. This is what I'm using based on this branch:


```rust
#[derive(Debug, Error)]
pub enum BlockError {
    #[error(display = "failed to get metadata of device fd")]
    Metadata(#[error(cause)] io::Error),
    #[error(display = "failed to reload partition table of device")]
    RereadTable(#[error(cause)] nix::Error),
}

pub fn reread_partition_table(file: &mut File) -> Result<(), BlockError> {
    #[cfg(target_os = "linux")]
    {
        use std::os::linux::fs::MetadataExt;
        use std::os::unix::io::AsRawFd;
        use gptman::linux::blkrrpart;

        const S_IFMT: u32 = 0o170_000;
        const S_IFBLK: u32 = 0o60_000;

        let metadata = file.metadata().map_err(BlockError::Metadata)?;
        if metadata.st_mode() & S_IFMT == S_IFBLK {
            if let Err(why) = unsafe { blkrrpart(file.as_raw_fd()) } {
                return Err(BlockError::RereadTable(why));
            }
        }
    }

    Ok(())
}
```

**Comment by cecton** [comment]
Do you think it's ok to wrap an unsafe function in a safe function in this case?

**Comment by mmstick** [comment]
Perhaps if we can provide a dependent type as an input that identifies that the input file is a device?

**Comment by cecton** [comment]
Please have a look again and let me know if this looks better. I'm also handling the case where the return code is non-zero. (I read somewhere in the Linux sources (I think) that the return code must be 0.)

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+os:
+  - linux
+  - windows
 
 rust:
   - stable
 
+# NOTE: disable Travis' cache for now because it makes the Windows build fail
+cache:
+  directories:
```
Comment (jens1o) [inline .travis.yml]: out of curiosity: Why is that?

**Comment by mmstick** [comment]
This works pretty well

File: .travis.yml (hunk 1 above)
Comment (cecton) [reply .travis.yml]: There is a timeout on the Windows build. The whole process succeed except at the end when it tries to generate the cache. Seeing how slow this VM is, I suspect it just takes too much time to compress https://travis-ci.org/cecton/gptman/jobs/555039669

Comment (jens1o) [reply .travis.yml]: Okay, thanks for the reply! :)

**Pull request merged**